src/richie/apps/search/elasticsearch.py

```diff
@@ -24,9 +26,13 @@ def __init__(self, hosts=None, transport_class=Transport, **kwargs):
         of Elasticsearch we're working with.
         """
         super().__init__(hosts=hosts, transport_class=transport_class, **kwargs)
-        # Store the ES version information in memory so we don't have to re-make the
-        # info request constantly.
-        self.__es_version__ = self.info()["version"]["number"][:1]
+
+    @cached_property
+    def __es_version__(self):
```

Comment:
Cached_property is thread safe so I'm pretty sure this would make the query for each incoming request. You should set it yourself in the client class. Would be nice if a test can evidence that 😏

Reply:
I think you would say `@cached_property` is not thead safe. I'll try to test this point and update the code accordingly.

Reply:
I'll make some successive autocomplete request that use `self.__es_version__` and info is not trigger on each request.